### PR TITLE
SIMPLY-2568: Show book progress below reader.

### PR DESCRIPTION
**Pull request by twaddington**
**What's this do?**
Show reading progress in Readium 2 viewer.

**Why are we doing this? (w/ JIRA link if applicable)**
https://jira.nypl.org/browse/SIMPLY-2568

**Screens**
[Screenshot_1582188868](https://user-images.githubusercontent.com/107117/74918291-ba4f5480-537d-11ea-9216-7a45187677b8.png)

### CODE REVIEW

File: org.librarysimplified.r2.views/src/main/java/org/librarysimplified/r2/views/SR2ReaderFragment.kt [hunk 1]
```diff
@@ -101,57 +107,70 @@ class SR2ReaderFragment : Fragment() {
...
+    logger.error("onBookOpenFailed: ", e)
+    runOnUIThread { onBookOpenFailedUI(e) }
+  }
 
-    this.runOnUIThread {
-      this.onBookOpenFailedUI(exception)
-    }
+  @UiThread
+  private fun onBookOpenFailedUI(e: Throwable) {
+    TODO()
```
Comment (io7m) [inline org.librarysimplified.r2.views/src/main/java/org/librarysimplified/r2/views/SR2ReaderFragment.kt]: Not sure what's going on here.

File: org.librarysimplified.r2.views/src/main/java/org/librarysimplified/r2/views/SR2ReaderFragment.kt [hunk 2]
```diff
@@ -101,57 +107,70 @@ class SR2ReaderFragment : Fragment() {
...
 
   private fun runOnUIThread(f: () -> Unit) {
     val looper = Looper.getMainLooper()
     val h = Handler(looper)
-    h.post {
-      f.invoke()
-    }
-  }
-
-  override fun onStop() {
```
Comment (io7m) [inline org.librarysimplified.r2.views/src/main/java/org/librarysimplified/r2/views/SR2ReaderFragment.kt]: We don't disconnect the view on stopping anymore?

**Comment by io7m** [comment]
To be clear, I don't see either of these as showstoppers. I just want to understand what the intention is.

File: org.librarysimplified.r2.views/src/main/java/org/librarysimplified/r2/views/SR2ReaderFragment.kt (hunk 1 above)
Comment (twaddington) [reply org.librarysimplified.r2.views/src/main/java/org/librarysimplified/r2/views/SR2ReaderFragment.kt]: I think this method was just empty before. The only change I made was add the "TODO()" but github is struggling with the diff.

File: org.librarysimplified.r2.views/src/main/java/org/librarysimplified/r2/views/SR2ReaderFragment.kt (hunk 2 above)
Comment (twaddington) [reply org.librarysimplified.r2.views/src/main/java/org/librarysimplified/r2/views/SR2ReaderFragment.kt]: We do! This method moved up to line 122-126. I'm used to grouping lifecycle methods together so it was moved up to be just after "onStop()".

**Comment by twaddington** [comment]
@io7m responded to your comments and rebased on develop.

File: org.librarysimplified.r2.views/src/main/java/org/librarysimplified/r2/views/SR2ReaderFragment.kt (hunk 1 above)
Comment (io7m) [reply org.librarysimplified.r2.views/src/main/java/org/librarysimplified/r2/views/SR2ReaderFragment.kt]: Right!

File: org.librarysimplified.r2.views/src/main/java/org/librarysimplified/r2/views/SR2ReaderFragment.kt (hunk 2 above)
Comment (io7m) [reply org.librarysimplified.r2.views/src/main/java/org/librarysimplified/r2/views/SR2ReaderFragment.kt]: :+1:

**Pull request merged**